library/Fission/Internal/Fixture/Bearer/Nested.hs

```diff
@@ -0,0 +1,57 @@
+{-# OPTIONS_GHC -fno-warn-incomplete-uni-patterns #-}
+
+-- | A real world example with a nested proof
+module Fission.Internal.Fixture.Bearer.Nested
+  ( jsonRSA2048
+  , tokenRSA2048
+  , jwtRSA2048
+  , rawContent
+  , validTime
+  , InTimeBounds (..)
+  ) where
+
+import           Servant.API
+
+
+import           Fission.Authorization.ServerDID.Class
+import qualified Fission.Internal.Time                 as Time
+import           Fission.Prelude
+
+import qualified Fission.Web.Auth.Token.Bearer.Types   as Bearer
+import           Fission.Web.Auth.Token.JWT            as JWT
+
+import           Fission.Web.Auth.Token.JWT.Resolver   as Proof
+
+newtype InTimeBounds a = InTimeBounds { unwrap :: Identity a }
+  deriving newtype
+    ( Functor
+    , Applicative
+    , Monad
+    , Eq
+    , Show
+    )
+
+instance Proof.Resolver InTimeBounds where
+  resolve = error "shouldn't hit this in our specific case"
+
+instance ServerDID InTimeBounds where
+  getServerDID = pure did
+    where
+      Right did = eitherDecode "\"did:key:zStEZpzSMtTt9k2vszgvCwF4fLQQSyA15W5AQ4z3AR6Bx4eFJ5crJFbuGxKmbma4\""
+
+instance MonadTime InTimeBounds where
+  currentTime = pure $ Time.fromSeconds 1591627760
```

Comment:
In case anyone is interested in how we test side effects in a totally pure way 😉